File: ls/server/src/main/java/org/opencds/cqf/cql/ls/server/utility/Futures.java
package org.opencds.cqf.cql.ls.server.utility;

import static com.google.common.base.Preconditions.checkNotNull;

import java.util.concurrent.CompletableFuture;

public class Futures {
    private Futures() {}

    public static <U> CompletableFuture<U> failed(Throwable ex) {
        checkNotNull(ex);
        CompletableFuture<U> future = new CompletableFuture<>();
        future.completeExceptionally(ex);
        return future;
    }
}

File: ls/server/src/main/kotlin/org/opencds/cqf/cql/ls/server/utility/Futures.kt
package org.opencds.cqf.cql.ls.server.utility

import java.util.concurrent.CompletableFuture

object Futures {
    fun <U> failed(ex: Throwable): CompletableFuture<U> = CompletableFuture<U>().also { it.completeExceptionally(ex) }
}
